src/System.Security.Cryptography.Algorithms/tests/HKDFTests.cs

```diff
@@ -0,0 +1,611 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.XUnitExtensions;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Algorithms.Tests
+{
+    public class HKDFTests
+    {
+        [Theory]
+        [MemberData(nameof(GetRfc5869TestCases))]
+        public void Rfc5869ExtractByteArrayTests(Rfc5869TestCase test)
+        {
+            byte[] prk = HKDF.Extract(test.Hash, test.Ikm, test.Salt);
+            Assert.Equal(test.Prk, prk);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRfc5869TestCases))]
+        public void Rfc5869ExtractByteArrayTamperHashTests(Rfc5869TestCase test)
+        {
+            byte[] prk = HKDF.Extract(HashAlgorithmName.MD5, test.Ikm, test.Salt);
+            Assert.NotEqual(test.Prk, prk);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRfc5869TestCases))]
+        public void Rfc5869ExtractByteArrayTamperIkmTests(Rfc5869TestCase test)
+        {
+            test.Ikm[0] ^= 1;
+            byte[] prk = HKDF.Extract(test.Hash, test.Ikm, test.Salt);
+            Assert.NotEqual(test.Prk, prk);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRfc5869TestCasesWithNonEmptySalt))]
+        public void Rfc5869ExtractByteArrayTamperSaltTests(Rfc5869TestCase test)
+        {
+            test.Salt[0] ^= 1;
+            byte[] prk = HKDF.Extract(test.Hash, test.Ikm, test.Salt);
+            Assert.NotEqual(test.Prk, prk);
+        }
+
+        [Fact]
+        public void Rfc5869ExtractByteArrayDefaultHash()
+        {
+            byte[] ikm = new byte[20];
+            byte[] salt = new byte[20];
+            Assert.Throws<ArgumentOutOfRangeException>(() => HKDF.Extract(default(HashAlgorithmName), ikm, salt));
+        }
+
+        [Fact]
+        public void Rfc5869ExtractByteArrayNonsensicalHash()
+        {
+            byte[] ikm = new byte[20];
+            byte[] salt = new byte[20];
+            Assert.Throws<ArgumentOutOfRangeException>(() => HKDF.Extract(new HashAlgorithmName("foo"), ikm, salt));
+        }
+
+        [Fact]
+        public void Rfc5869ExtractByteArrayNullIkm()
+        {
+            byte[] salt = new byte[20];
+            Assert.Throws<ArgumentNullException>(() => HKDF.Extract(HashAlgorithmName.SHA1, null, salt));
+        }
+
+        [Fact]
+        public void Rfc5869ExtractByteArrayEmptyIkm()
+        {
+            byte[] salt = new byte[20];
+            byte[] ikm = Array.Empty<byte>();
+
+            // Ensure does not throw
+            byte[] prk = HKDF.Extract(HashAlgorithmName.SHA1, ikm, salt);
+            Assert.Equal("FBDB1D1B18AA6C08324B7D64B71FB76370690E1D".HexToByteArray(), prk);
```

Comment:
Doing this the other way usually makes for nicer messages (since the array print uses decimal values):

```suggestion
            Assert.Equal("FBDB1D1B18AA6C08324B7D64B71FB76370690E1D", prk.ByteArrayToHex());
```